relay-monitors/src/lib.rs

```diff
@@ -176,7 +193,24 @@ pub fn process_check_in(payload: &[u8]) -> Result<Vec<u8>, ProcessCheckInError>
         return Err(ProcessCheckInError::InvalidEnvironment);
     }
 
-    Ok(serde_json::to_vec(&check_in)?)
+    static NAMESPACE: OnceCell<Uuid> = OnceCell::new();
+    let namespace = NAMESPACE
+        .get_or_init(|| Uuid::new_v5(&Uuid::NAMESPACE_URL, b"https://sentry.io/crons/#did"));
+
+    // Use the project_id + monitor_slug as the routing key hint. This helps ensure monitor
+    // check-ins are processed in order by consistently routing check-ins from the same monitor.
+    let project_id_slug_key: Vec<u8> = [
+        &project_id.value().to_be_bytes()[..],
+        check_in.monitor_slug.as_bytes(),
+    ]
+    .concat();
```

Comment:
Was struggling a bit to get this to work, not 100% sure if this is the most efficient implementation or not.

Feel free to update the PR if there is a better approach